scapy/arch/linux.py

```diff
@@ -600,7 +600,7 @@ def send(self, x):
         # type: (Packet) -> int
         iff = x.route()[0]
         if iff is None:
-            iff = conf.iface
+            iff = network_name(conf.iface)
```

Comment:
I suspect L2Socket needs a similar change (line 502) for the case where `iface` is None (line 479)

Reply:
No that's already taken care of line 479